pkgs/applications/display-managers/greetd/qtgreet/default.nix

```diff
@@ -0,0 +1,58 @@
+{
+  stdenv,
+  lib,
+
+  fetchFromGitLab,
+
+  wrapQtAppsHook,
+  meson,
+  pkg-config,
+  cmake,
+  qttools,
+  ninja,
+
+  qtbase,
+  json_c,
+  wlroots,
+  wayqt,
+  wayland,
+  pixman
+}:
+
+stdenv.mkDerivation (finalAtrrs: {
+  pname = "qtgreet";
+  version = "unstable-2022-01-24"; # Couldn't get 1.0.0 to build
+
+  src = fetchFromGitLab {
+    owner = "marcusbritanicus";
+    repo = finalAtrrs.pname;
```

Comment:
```suggestion
    repo = "qtgreet";
```

Reply:
I prefer it the other way.

Reply:
finalAttrs has a small performance penalty and is not necessary to use here.

Reply:
With:

```
hyperfine --warmup 1 -r 30 'nix-instantiate -A greetd.qtgreet'
Benchmark 1: nix-instantiate -A greetd.qtgreet
  Time (mean ± σ):     558.7 ms ±  12.3 ms    [User: 380.6 ms, System: 92.1 ms]
  Range (min … max):   531.6 ms … 575.0 ms    30 runs
```

Without:

```
hyperfine --warmup 1 -r 30 'nix-instantiate -A greetd.qtgreet'
Benchmark 1: nix-instantiate -A greetd.qtgreet
  Time (mean ± σ):     535.4 ms ±  20.0 ms    [User: 362.9 ms, System: 88.0 ms]
  Range (min … max):   509.1 ms … 578.5 ms    30 runs
```